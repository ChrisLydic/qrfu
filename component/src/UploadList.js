import React from 'react';
import PropTypes from 'prop-types';

// example components for displaying uploaded files

export default function UploadList(props) {
    const items = props.uploads.map(file => {
        if (file.type === 'audio') {
            return (<li key={file.name}>
                        <audio src={`${file.url}`} controls></audio>
                        <div>{file.name}</div>

                        {props.delete &&
                            <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
                        }
                    </li>);
        } else {
            return (<li key={file.name}>
                        <a href={`${file.url}`}>
                            <img src={`${file.url}`} alt='preview' />
                            <div>{file.name}</div>
                        </a>

                        {props.delete && 
                            <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
                        }
                    </li>);
        }
    });
    
    if (props.uploads.length > 1 && props.delete) {
        return (
            <div>
                <button className="qr-delete-all" type="button" onClick={() => {props.uploads.forEach(f => props.delete(f.name))}}>
                    Delete All
                </button>

                <ul className='custom'>{items}</ul>
            </div>
        );
    } else {
        return (
            <ul className='custom'>{items}</ul>
        );
    }
}

export function LinkList(props) {
    const items = props.uploads.map(file => 
        (<li key={file.name}>
            <a href={`${file.url}`}>
                {file.name}
            </a>
            {props.delete &&
                <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
            }
        </li>)
    );
    
    if (props.uploads.length > 1 && props.delete) {
        return (
            <div>
                <button className="qr-delete-all" type="button" onClick={() => {props.uploads.forEach(f => props.delete(f.name))}}>
                    Delete All
                </button>

                <ul>{items}</ul>
            </div>
        );
    } else {
        return (
            <ul>{items}</ul>
        );
    }
}

export function ProfileImage(props) {
    if (props.uploads.length > 0) {
        let file = props.uploads[props.uploads.length - 1];

        return (
            <ul className='custom'>
                <li key={file.name}>
                    <a href={`${file.url}`}>
                        <img src={`${file.url}`} alt='preview' />
                        <div>{file.name}</div>
                    </a>

                    {props.delete && 
                        <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
                    }
                </li>
            </ul>
        );
    } else {
        return (
            <ul className='custom'></ul>
        );
    }
}

UploadList.propTypes = LinkList.propTypes = ProfileImage.propTypes = {
    uploads: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            url: PropTypes.string.isRequired,
            uid: PropTypes.string,
            type: PropTypes.string,
        })
    ).isRequired,
    delete: PropTypes.func,
};