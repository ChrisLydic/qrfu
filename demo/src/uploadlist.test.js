import React from 'react';
import renderer from 'react-test-renderer';
import UploadList, { LinkList, ProfileImage } from './uploadlist';

test('List displays uploads in their specific formats', () => {
    const files = [
        {
            name: 'file-img',
            url: 'http://localhost:3001/',
            uid: '10001010100110',
            type: 'image',
        },
        {
            name: 'file-audio',
            url: 'http://localhost:3001/',
            uid: '00011010100110',
            type: 'audio',
        }
    ];

    const component = renderer.create(
        <UploadList uploads={files} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('List displays the file links', () => {
    const files = [
        {
            name: 'file-img',
            url: 'http://localhost:3001/',
            uid: '10001010100110',
            type: 'image',
        },
        {
            name: 'file-audio',
            url: 'http://localhost:3001/',
            uid: '00011010100110',
            type: 'audio',
        }
    ];

    const component = renderer.create(
        <LinkList uploads={files} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Profile image displays first file', () => {
    const files = [
        {
            name: 'file-img',
            url: 'http://localhost:3001/',
            uid: '10001010100110',
            type: 'image',
        },
        {
            name: 'file-2',
            url: 'http://localhost:3001/',
            uid: '00011010100110',
            type: 'image',
        }
    ];

    const component = renderer.create(
        <ProfileImage uploads={files} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Profile image is set to default empty tag', () => {
    const files = [];

    const component = renderer.create(
        <ProfileImage uploads={files} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});