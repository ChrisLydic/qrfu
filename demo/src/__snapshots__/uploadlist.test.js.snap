// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List displays the file links 1`] = `
<ul>
  <li>
    <a
      href="http://localhost:3001/"
    >
      file-img
    </a>
  </li>
  <li>
    <a
      href="http://localhost:3001/"
    >
      file-audio
    </a>
  </li>
</ul>
`;

exports[`List displays uploads in their specific formats 1`] = `
<ul
  className="custom"
>
  <li>
    <a
      href="http://localhost:3001/"
    >
      <img
        alt="preview"
        src="http://localhost:3001/"
      />
      <div>
        file-img
      </div>
    </a>
  </li>
  <li>
    <audio
      controls={true}
      src="http://localhost:3001/"
    />
    <div>
      file-audio
    </div>
  </li>
</ul>
`;

exports[`Profile image displays first file 1`] = `
<span>
  <img
    alt="avatar"
    className="profilePic"
    src="http://localhost:3001/"
  />
</span>
`;

exports[`Profile image is set to default empty tag 1`] = `
<div
  className="profilePic"
/>
`;
